src/NEO/Address.cpp

```diff
@@ -0,0 +1,70 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "Address.h"
+
+#include "../Hash.h"
+
+#include <TrezorCrypto/base58.h>
+
+using namespace TW::NEO;
+
+bool Address::isValid(const std::string& string) {
+    size_t capacity = 128;
+    uint8_t buffer[capacity];
+
+    int size = base58_decode_check(string.data(), HASHER_SHA2D, buffer, (int)capacity);
+    if (size != Address::size || buffer[0] != version) {
+        return false;
+    }
+
+    return true;
+}
+
+Address::Address(const std::string& string) {
+    size_t capacity = 128;
+    uint8_t buffer[capacity];
+
+    int size = base58_decode_check(string.data(), HASHER_SHA2D, buffer, (int)capacity);
+    assert(size == Address::size);
```

Comment:
Let's not use assert, look at the latest code on master.

Reply:
done